### PR TITLE
Fix BIOPET-379

### PR DESCRIPTION
**Pull request by ffinfo**

### CODE REVIEW

File: biopet-core/src/main/scala/nl/lumc/sasc/biopet/core/MultiSampleQScript.scala [hunk 1]
```diff
@@ -141,11 +141,17 @@ trait MultiSampleQScript extends SummaryQScript { qscript: QScript =>
...
     /** returns a set with library names or throws error when not found */
     protected def libIds: Set[String] = {
-      val ids = ConfigUtils.getMapFromPath(globalConfig.map, List("samples", sampleId, "libraries")).getOrElse(Map()).keySet
-      if (ids.isEmpty) {
-        throw new IllegalStateException(s"No libraries found in config for sample $sampleId")
+      val ids: Set[String] = try {
+        ConfigUtils.getMapFromPath(globalConfig.map, List("samples", sampleId, "libraries")).getOrElse(Map()).keySet
+      } catch {
+        case e: IllegalStateException if e.getMessage == "Value is not a map: library" =>
+          Logging.addError("libraries is not a map")
```
Comment (sndrtj) [inline biopet-core/src/main/scala/nl/lumc/sasc/biopet/core/MultiSampleQScript.scala]: This is a somewhat nebulous error message if one is not familiar with the internal structure we use here. Perhaps simply state the contents are wrongly formatted?

File: biopet-core/src/main/scala/nl/lumc/sasc/biopet/core/BiopetQScript.scala [hunk 2]
```diff
@@ -177,4 +177,52 @@ object BiopetQScript {
...
+
+  def safeFailFiles(function: QFunction): Seq[File] = {
+    try {
+      function.failOutputs
+    } catch {
+      case e: NullPointerException => Seq()
+    }
+  }
+
+  def safeIsDone(function: QFunction): Boolean = {
```
Comment (sndrtj) [inline biopet-core/src/main/scala/nl/lumc/sasc/biopet/core/BiopetQScript.scala]: I would change this function and `safeIsFail` to have type `Option[Boolean]`. You return simply `false` here when the underlying method does not exist, but I believe that's potentially giving a wrong answer. Returning `None` (as in, no boolean exists here) is semantically better, imho.

File: biopet-core/src/test/scala/nl/lumc/sasc/biopet/core/MultiSampleQScriptTest.scala [hunk 3]
```diff
@@ -272,7 +276,7 @@ object MultiSampleQScriptTest {
       def summaryFiles: Map[String, File] = Map()
 
       /** Name of summary output file */
-      def summaryFile: File = null
+      def summaryFile: File = new File("./summary.json")
```
Comment (sndrtj) [inline biopet-core/src/test/scala/nl/lumc/sasc/biopet/core/MultiSampleQScriptTest.scala:279]: If this file gets created during the test, we want to remove it after teardown.

**Review by sndrtj — CHANGES_REQUESTED** [review]
I have several comments you may want to look at on this PR.

File: biopet-core/src/test/scala/nl/lumc/sasc/biopet/core/MultiSampleQScriptTest.scala (hunk 3 above)
Comment (ffinfo) [reply biopet-core/src/test/scala/nl/lumc/sasc/biopet/core/MultiSampleQScriptTest.scala:279]: is does not get created :)

File: biopet-core/src/main/scala/nl/lumc/sasc/biopet/core/BiopetQScript.scala (hunk 2 above)
Comment (ffinfo) [reply biopet-core/src/main/scala/nl/lumc/sasc/biopet/core/BiopetQScript.scala]: maybe indeed a better option

File: biopet-core/src/main/scala/nl/lumc/sasc/biopet/core/MultiSampleQScript.scala (hunk 1 above)
Comment (ffinfo) [reply biopet-core/src/main/scala/nl/lumc/sasc/biopet/core/MultiSampleQScript.scala]: changed this

File: biopet-core/src/main/scala/nl/lumc/sasc/biopet/core/BiopetQScript.scala (hunk 2 above)
Comment (ffinfo) [reply biopet-core/src/main/scala/nl/lumc/sasc/biopet/core/BiopetQScript.scala]: Changed this

**Review by sndrtj — APPROVED** [review]

**Pull request merged**